lib/ansible/executor/task_executor.py

```diff
@@ -776,10 +777,17 @@ def _get_action_handler(self, connection, templar):
         Returns the correct action plugin to handle the requestion task action
         '''
 
+        module_path = self._shared_loader_obj.module_loader.find_plugin(self._task.action)
         module_prefix = self._task.action.split('_')[0]
+        is_python_module = module_path.endswith('.py')
```

Comment:
To be  generic, there needs to be a loop in here somewhere.  I'm guessing that find_plugin() returns the "best" fit for the plugin to use and we need to add PluginLoader API which returns all the plugins that match the name of the action.  then loop over those looking for a .py file.

Also, we can't rely on extension here because modules might have no extension.  Instead we have to try to parse it and if it fails, try the next potentially matching module.